src/BloomBrowserUI/pageChooser/js/page-chooser.js

```diff
@@ -43,6 +44,40 @@ var PageChooser = (function () {
         $("#preview").attr("src", $(this._selectedGridItem).find("img").first().attr("src"));
         this.setLocalizedText($('#previewDescriptionText'), 'TemplateBooks.PageDescription.', $(".pageDescription", this._selectedGridItem).text(), defaultCaptionText);
     }; // thumbnailClickHandler
+    // There's a bug deep in javascript that doesn't take into account the scrolling
+    // of a div element before something inside it is clicked on.  The following code
+    // detects whether the scrolling has changed since the last mouse click, and if so,
+    // searches for the item which should have matched.  For the initial bug report,
+    // see https://silbloom.myjetbrains.com/youtrack/issue/BL-2623.
+    // Note that the offset().top values returned by jquery properly take into account
+    // the scrollTop of the scrolling parent div.  Which makes me think the bug may be
+    // below the jquery level!?
+    PageChooser.prototype.findProperElement = function (clickedDiv, evt) {
+        var gridItem = $(clickedDiv).parent();
+        if (evt) {
+            var scrollingParent = $(clickedDiv).parentsUntil("div.bloomDialogContainer", "div.gridItemDisplay");
```

Comment:
It feels rather weird that we go searching for the parent scrolling div (which suggests there might be more than one) but we only have one place to store _scrollTop (which means there had better be only one scrolling parent).
In general, various names suggest this is generic code, but things like knowing to go two layers up from clickedDiv to container make it quite specific to this dialog. Not sure how to improve it, but I'll let you think it over. For example, could we initialize a member _scrollingPageList by ID and then have _scrollTopOfScrollingPageList?
Then again, maybe we'll want something like this somewhere else, and your version is at least half way to being generic. Might be best to just leave it. Anyway, I'll let you think it over.